code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -0,0 +1,376 @@
+#define IC_COMPONENTS_BASE		25
+#define IC_COMPLEXITY_BASE		75
+
+/obj/item/device/electronic_assembly
+	name = "electronic assembly"
+	desc = "It's a case, for building small electronics with."
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/electronic_assemblies.dmi'
+	icon_state = "setup_small"
+	flags_1 = NOBLUDGEON_1
+	var/max_components = IC_COMPONENTS_BASE
+	var/max_complexity = IC_COMPLEXITY_BASE
+	var/opened = FALSE
+	var/obj/item/stock_parts/cell/battery // Internal cell which most circuits need to work.
+	var/cell_type = /obj/item/stock_parts/cell
+	var/can_charge = TRUE //Can it be charged in a recharger?
+	var/charge_sections = 4
+	var/charge_tick = FALSE
+	var/charge_delay = 4
+	var/use_cyborg_cell = TRUE
+
+/obj/item/device/electronic_assembly/medium
+	name = "electronic mechanism"
+	icon_state = "setup_medium"
+	desc = "It's a case, for building medium-sized electronics with."
+	w_class = WEIGHT_CLASS_SMALL
+	max_components = IC_COMPONENTS_BASE * 2
+	max_complexity = IC_COMPLEXITY_BASE * 2
+
+/obj/item/device/electronic_assembly/large
+	name = "electronic machine"
+	icon_state = "setup_large"
+	desc = "It's a case, for building large electronics with."
+	w_class = WEIGHT_CLASS_BULKY
+	max_components = IC_COMPONENTS_BASE * 4
+	max_complexity = IC_COMPLEXITY_BASE * 4
+	anchored = 0
+
+/obj/item/device/electronic_assembly/large/attackby(var/obj/item/O, var/mob/user)
+	if(istype(O, /obj/item/wrench))
+		if(!isturf(loc))
+			return
+		anchored = !anchored
+		to_chat(user,"You [anchored ? "wrench" : "unwrench"] \the [src].")
+		return
+	..()
```

Comment:
no . = ..()? (all these procs, not just this one.)

Reply:
not sure,if it is good idea.because this way you may block several types of interaction.